src/shared/util.js

```diff
@@ -79,15 +79,31 @@ export function isPromise (val: any): boolean {
   )
 }
 
+export function getTrustedTypes() {
+  // TrustedTypes have been renamed to trustedTypes https://github.com/WICG/trusted-types/issues/177
+  return typeof window !== 'undefined' && (window.trustedTypes || window.TrustedTypes);
+}
+
+export function isTrustedValue(value: any): boolean {
+  const tt = getTrustedTypes();
+  if (!tt) return false;
```

Comment:
Is `window.trustedTypes` only present when the relevant CSP is enabled?

Reply:
No, it will be available natively in browsers (for now just chrome with `Experimental Web Platform features` enabled) or when there is a [polyfill](https://github.com/WICG/trusted-types).

For this function it's logical to return `false` when Trusted Types are not available. However, I think you meant the case when application is running in environment that has Trusted Types, but doesn't want to use them. In this case, the application won't have the CSP header present and in that case Trusted Types act as noop. However, we might add a parameter to Trusted Types to tell if the application wants to use them or not. This is already [tracked issue](https://github.com/WICG/trusted-types/issues/36).

Reply:
EDIT: The code before was not compliant with what I said as I would throw and error in a specific case (innerHTML in svg in IE with trustedTypes available). I've fixed the code not to throw in this case.